Assets/Scripts/MeleeController.cs

```diff
@@ -0,0 +1,34 @@
+using UnityEngine;
+
+public class MeleeController : MonoBehaviour
+{
+    public Transform slashPoint;
+    public GameObject slashPrefab;
+    
+    float slashCooldown = 0.6f; //Upgradable value
+    float timeLeft;
+    bool canSlash = true;
+
+    void Update()
+    {
+        if(timeLeft > 0)
+        {
+            canSlash = false;
+            timeLeft -= Time.deltaTime;
+        }
+        else
+        {
+            canSlash = true;
+        }
+        if(Input.GetButtonDown("Fire2") && canSlash)
+        {
+            Swing();
+            timeLeft = slashCooldown;
+        }
+        
+    }
+    void Swing()
+    {
+        GameObject bullet = Instantiate(slashPrefab, slashPoint.position, slashPoint.rotation * Quaternion.Euler(0, 0, 90));
```

Comment:
Why is it bullet if its melee controller?

Reply:
Whoops, copy typo